zcash_client_sqlite/src/testing.rs

```diff
@@ -1061,3 +1070,15 @@ pub(crate) fn input_selector(
     let change_strategy = standard::SingleOutputChangeStrategy::new(fee_rule, change_memo);
     GreedyInputSelector::new(change_strategy, DustOutputPolicy::default())
 }
+
+pub(crate) fn check_proposal_serialization_roundtrip(
```

Comment:
Add documentation to this method noting that this is not actually checking the on-the-wire serialization (as protobuf [does not provide a canonical encoding](https://github.com/protocolbuffers/protobuf/issues/3417)), but is just checking the round-trip of the "structural encoding" to the protobuf encoder's types.